includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -2843,4 +2846,33 @@ function( $cost, $qty ) {
 			'discount_amount'     => array_sum( array_column( $items, 'discount_amount' ) ),
 		];
 	}
+
+	/**
+	 * Move the email field to the top of the Checkout page.
+	 *
+	 * @param array $fields WooCommerce checkout fields.
+	 *
+	 * @return array WooCommerce checkout fields.
+	 */
+	public function checkout_update_email_field_priority( $fields ) {
+		$is_link_enabled = in_array(
+			Link_Payment_Method::PAYMENT_METHOD_STRIPE_ID,
+			\WC_Payments::get_gateway()->get_payment_method_ids_enabled_at_checkout( null, true ),
+			true
+		);
+
+		if ( $is_link_enabled ) {
+			// Update the field priority.
+			$fields['billing_email']['priority'] = 1;
+
+			// Add extra `wcpay-checkout-email-field` class.
+			$fields['billing_email']['class'][] = 'wcpay-checkout-email-field';
+
+			// Append StripeLink modal trigger button for logged in users.
+			$fields['billing_email']['label'] = $fields['billing_email']['label']
+				. ' <span class="wcpay-stripelink-trigger-modal">&nbsp;</span>';
```

Comment:
I wonder if this span's class should be renamed to reflect that it's not the modal itself but a "button that opens the modal.